Tests/RegexTests/MatchTests.swift

```diff
@@ -1046,14 +1053,31 @@ extension RegexTests {
     firstMatchTest(
       #"(?:a|.b)c"#, input: "123abcacxyz", match: "abc")
     firstMatchTest(
-      #"(?>a|.b)c"#, input: "123abcacxyz", match: "ac", xfail: true)
+      #"(?>a|.b)c"#, input: "123abcacxyz", match: "ac")
     firstMatchTest(
-      "(*atomic:a|.b)c", input: "123abcacxyz", match: "ac", xfail: true)
+      "(*atomic:a|.b)c", input: "123abcacxyz", match: "ac")
     firstMatchTest(
       #"(?:a+)[a-z]c"#, input: "123aacacxyz", match: "aac")
     firstMatchTest(
-      #"(?>a+)[a-z]c"#, input: "123aacacxyz", match: "ac", xfail: true)
+      #"(?>a+)[a-z]c"#, input: "123aacacxyz", match: nil)
+    
+    // Atomicity should stay in the atomic group
+    firstMatchTest(
+      #"(?:(?>a)|.b)c"#, input: "123abcacxyz", match: "abc")
 
+    // Quantifier behavior inside atomic
+    firstMatchTest(
+      #"^(?>a+?)a$"#, input: "aa", match: "aa")
+    firstMatchTest(
+      #"^(?>a+?)a$"#, input: "aaa", match: nil)
+    firstMatchTest(
+      #"(?>a++)a"#, input: "aaa", match: nil)
+
+    firstMatchTest(
+      #"(?>(\d+))\w+\1"#, input: "123x12", match: nil)
+    firstMatchTest(
+      #"(?>(\d+))\w+\1"#, input: "123x23", match: "23x23",
+      xfail: true)
```

Comment:
Could you add positive tests and non-atomic versions?

Reply:
Noting that for these or all of them, we can also do whole-match semantics, which might be clearer. Or match-from-front, in which case we'd see what alternative was taken